pkg/antctl/raw/featuregates/command_test.go

```diff
@@ -0,0 +1,400 @@
+				return restClient, nil
+			},
+		},
+		{
+			name:           "get featuregates in agent Pod",
+			runE:           agentRunE,
+			expectedOutput: "Antrea Agent Feature Gates\nFEATUREGATE              STATUS         VERSION   \nEndpointSlice            Disabled       ALPHA     \nAntreaIPAM               Disabled       ALPHA     \nFlowExporter             Disabled       ALPHA     \nNetworkPolicyStats       Enabled        BETA      \nAntreaPolicy             Enabled        BETA      \nMulticast                Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nAntreaProxy              Enabled        BETA      \nEgress                   Enabled        BETA      \nNodePortLocal            Enabled        BETA      \nMulticluster             Enabled        ALPHA     \n",
+			getFakeRestClient: func(kubeconfig *rest.Config, k8sClientset kubernetes.Interface, antreaClientset antrea.Interface, mode string) (*rest.RESTClient, error) {
+				fakeHttpClient := fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+					return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(bytes.NewBuffer(agentResponse))}, nil
+				})
+				restClient.Client = fakeHttpClient
+				return restClient, nil
+			},
+		},
+		{
+			name:           "get featuregates in controller Pod",
+			runE:           controllerLocalRunE,
+			expectedOutput: "Antrea Agent Feature Gates\nFEATUREGATE              STATUS         VERSION   \nEndpointSlice            Disabled       ALPHA     \nAntreaIPAM               Disabled       ALPHA     \nFlowExporter             Disabled       ALPHA     \nNetworkPolicyStats       Enabled        BETA      \nAntreaPolicy             Enabled        BETA      \nMulticast                Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nAntreaProxy              Enabled        BETA      \nEgress                   Enabled        BETA      \nNodePortLocal            Enabled        BETA      \nMulticluster             Enabled        ALPHA     \nAntrea Controller Feature Gates\nFEATUREGATE              STATUS         VERSION   \nAntreaPolicy             Enabled        BETA      \nNodeIPAM                 Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nEgress                   Enabled        BETA      \nNetworkPolicyStats       Enabled        BETA      \n",
+			getFakeRestClient: func(kubeconfig *rest.Config, k8sClientset kubernetes.Interface, antreaClientset antrea.Interface, mode string) (*rest.RESTClient, error) {
+				fakeHttpClient := fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+					return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(bytes.NewBuffer(fullResponse))}, nil
+				})
+				restClient.Client = fakeHttpClient
+				return restClient, nil
+			}},
```

Comment:
you are repeating 3 times the same code I think? If you use the response as the test parameter, you can build the fake function in the test body and avoid the duplication.

Reply:
Refactored. thanks.